### PR TITLE
creating TaskDetails content

### PR DESCRIPTION
**Pull request by JonathanYesh279**

### CODE REVIEW

File: src/assets/styles/components/task-details.scss [hunk 1]
```diff
@@ -1,5 +1,69 @@
...
+      width: 100%;
+    }
+  
+    .title-input {
+      display: inline-flex;
+      font-size: 20px;
+      background: transparent;
+      width: 88%;
+      color: var(--ds-text);
+      padding: 2px 2px;
```
Comment (tomercarolla) [inline src/assets/styles/components/task-details.scss:25]: padding: 2px;

File: src/assets/styles/components/task-details.scss [hunk 2]
```diff
@@ -1,5 +1,69 @@
...
+      .btn-container {
+        display: flex;
+        width: 100%;
+
+        .nav-btn {
+          display: flex;
+          justify-content: flex-start;
+          background: transparent;
+          border: none;
+          margin: 5px 5px;
```
Comment (tomercarolla) [inline src/assets/styles/components/task-details.scss:61]: margin: 5px;

File: src/assets/styles/components/task-details.scss [hunk 3]
```diff
@@ -1,5 +1,69 @@
...
+    p {
+      padding:0;
+      margin:5px 0 0 75px;
+      width: 100%;
+    }
+  }
+
+  .task-body {
+    display: grid;
+    grid-template-columns: 3fr 1fr;
```
Comment (tomercarolla) [inline src/assets/styles/components/task-details.scss:45]: the grid dont looks, right. also please check the responsive in trello.

**Review by tomercarolla — CHANGES_REQUESTED** [review]
image

you have a lot of errors when open the dialog - please check it

**Comment by tomercarolla** [comment]
@JonathanYesh279 
image

change the folder name task-details

File: src/assets/styles/components/task-details.scss (hunk 3 above)
Comment (tomercarolla) [reply src/assets/styles/components/task-details.scss:45]: @JonathanYesh279 I think you need to add responsive, look the dialog in trello. under 750px
image

File: src/ui/Buttons/Button.jsx [hunk 4]
```diff
@@ -15,7 +15,7 @@ export const Button = styled.button`
   font-size: 14px;
   font-weight: 500;
   background-color: var(--fill, var(--ds-background-brand-bold));
-  color: var(--text, var(--ds-text-inverse));
+  color: var(--text), var(--ds-text-inverse));
```
Comment (tomercarolla) [inline src/ui/Buttons/Button.jsx:18]: `color: var(--text), var(--ds-text-inverse));`

you have here an error. check it

Comment (JonathanYesh279) [reply src/ui/Buttons/Button.jsx:18]: yea, I've added here parentheses, it was missing, i don't know why..

File: src/assets/styles/components/task-details.scss (hunk 3 above)
Comment (JonathanYesh279) [reply src/assets/styles/components/task-details.scss:45]: I've added a media queris to this container and made it responsive under 750px, make it two columns at the bottom of the page.

File: src/assets/styles/components/nav-taskdetails.scss [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+@media (max-width: 750px) {
+  .nav-task-body {
+    display: grid;
+    grid-template-columns: 1fr 1fr;
+    grid-column: 1 / -1;
+    margin-top: 20px;
+  }
+}
```
Comment (tomercarolla) [inline src/assets/styles/components/nav-taskdetails.scss:34]: move inside the class.

```
.nav-task-body{
....

@media (max-width: 750px) {
  ....
  }
}
}
```

File: src/assets/styles/components/task-details.scss (hunk 3 above)
Comment (tomercarolla) [reply src/assets/styles/components/task-details.scss:52]: move the media query inside the class. look my comment above

File: src/assets/styles/components/task-details.scss [hunk 6]
```diff
@@ -1,5 +1,143 @@
...
+      }
+
+      .description-container {
+        margin-top: 20px;
+
+        .btn-description {
+          display: flex;
+          align-items: flex-start;
+          color: var(--ds-text);
+          padding: 10px 0 0 10px;
```
Comment (tomercarolla) [inline src/assets/styles/components/task-details.scss:92]: fix the padding. paddting-top: 10px;
padding-left: 10px;

File: src/assets/styles/components/task-details.scss [hunk 7]
```diff
@@ -1,5 +1,143 @@
...
+          width: 100%;
+          height: 8vh;
+          margin: 15px 0 0 30px;
```
Comment (tomercarolla) [inline src/assets/styles/components/task-details.scss:95]: fix the margin

File: src/assets/styles/main.scss [hunk 8]
```diff
@@ -23,3 +23,4 @@
 @import "components/workspace-content";
 @import "components/modal";
 @import "components/task-details";
+@import "components/nav-taskdetails";
```
Comment (tomercarolla) [inline src/assets/styles/main.scss:26]: name not good. fix it please

File: src/components/task-details/TaskDetails.jsx [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+            </div>
+
+            <div className="actions">
+              <span>Notifications</span>
+              <Button
+                scale="neutral"
+                fullwidth={true}
+                className="btn-notification"
+              >
+                <Icon name="watch" size="16px" scale="ghost" />
```
Comment (tomercarolla) [inline src/components/task-details/TaskDetails.jsx:91]: icon dont have scale

File: src/components/task-details/components/NavTaskDetails.jsx [hunk 10]
```diff
@@ -0,0 +1,100 @@
+import { Button, Icon } from "@ui";
+
+export function NavTaskDetails() {
+  return (
+    <nav className="nav-task-body">
+      <div className="btn-container">
+        <Button scale="neutral" className="btn-nav">
+          <Icon name="join" size="18px" />
+          Join
```
Comment (tomercarolla) [inline src/components/task-details/components/NavTaskDetails.jsx:9]: add span <span>Join</span>

File: src/components/task-details/TaskDetails.jsx [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+                Watch
```
Comment (tomercarolla) [inline src/components/task-details/TaskDetails.jsx:92]: add span

File: src/components/task-details/components/NavTaskDetails.jsx [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+        </Button>
+      </div>
+
+      <div className="btn-container">
+        <Button scale="neutral" className="btn-nav">
+          <Icon name="member" size="18px" />
+          Members
```
Comment (tomercarolla) [inline src/components/task-details/components/NavTaskDetails.jsx:16]: add span

File: src/components/task-details/components/NavTaskDetails.jsx [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+        </Button>
+      </div>
+
+      <div className="btn-container">
+        <Button scale="neutral" className="btn-nav">
+          <Icon name="label" size="18px" />
+          Labels
```
Comment (tomercarolla) [inline src/components/task-details/components/NavTaskDetails.jsx:23]: add span

File: src/components/task-details/components/NavTaskDetails.jsx [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+        <Button scale="neutral" className="btn-nav">
+          <Icon name="date" size="18px" />
+          Dates
+        </Button>
+      </div>
+
+      <div className="btn-container">
+        <Button scale="neutral" className="btn-nav">
+          <Icon name="attachment" size="18px" />
+          Attachments
```
Comment (tomercarolla) [inline src/components/task-details/components/NavTaskDetails.jsx:44]: add span

File: src/components/task-details/components/NavTaskDetails.jsx [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+        </Button>
+      </div>
+
+      <div className="btn-container">
+        <Button scale="neutral" className="btn-nav">
+          <Icon name="cover" size="18px" />
+          Cover
```
Comment (tomercarolla) [inline src/components/task-details/components/NavTaskDetails.jsx:51]: add span

File: src/components/task-details/components/NavTaskDetails.jsx [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+        </Button>
+      </div>
+
+      <div className="btn-container">
+        <Button scale="neutral" className="btn-nav">
+          <Icon name="customFields" size="18px" />
+          CustomFields
```
Comment (tomercarolla) [inline src/components/task-details/components/NavTaskDetails.jsx:58]: add span and fix the text

File: src/components/task-details/components/NavTaskDetails.jsx [hunk 17]
```diff
@@ -0,0 +1,100 @@
...
+        </Button>
+      </div>
+
+      <span>Actions</span>
+
+      <div className="btn-container">
+        <Button scale="neutral" className="btn-nav">
+          <Icon name="move" size="18px" />
+          Move
```
Comment (tomercarolla) [inline src/components/task-details/components/NavTaskDetails.jsx:67]: add span

File: src/components/task-details/components/NavTaskDetails.jsx [hunk 18]
```diff
@@ -0,0 +1,100 @@
...
+        </Button>
+      </div>
+
+      <div className="btn-container">
+        <Button scale="neutral" className="btn-nav">
+          <Icon name="copy" size="18px" />
+          Copy
```
Comment (tomercarolla) [inline src/components/task-details/components/NavTaskDetails.jsx:74]: add span

File: src/components/task-details/components/NavTaskDetails.jsx
Comment (tomercarolla) [inline src/components/task-details/components/NavTaskDetails.jsx:1]: please refactor this code. use object you repeat yourself a lot

File: src/components/task-details/TaskDetails.jsx (hunk 9 above)
Comment (JonathanYesh279) [reply src/components/task-details/TaskDetails.jsx:91]: what do you mean? It has scale="ghost". What should it get?

File: src/assets/styles/components/sidebar-taskdetails.scss [hunk 19]
```diff
@@ -0,0 +1,51 @@
+.nav-task-body {
```
Comment (tomercarolla) [inline src/assets/styles/components/sidebar-taskdetails.scss:1]:
```
.nav-task-body {
  display: flex;
  flex-direction: column;

@media (max-width: 750px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-column: 1 / -1;
    margin-top: 20px;

    span {
      grid-column: 1 / -1;
      display: block;
      margin-bottom: 10px;
    }

     .btn-container {
      .btn-nav {
        align-items: center; 
      }
    }
}

  span {
    font-size: 12px;
    margin-top: 10px;
  }

  .btn-container {
    display: flex;
    width: 100%;

    span {
      display: flex;
      font-size: 14px;
    }

    .btn-nav {
      display: flex;
      align-items: flex-end;
      justify-content: flex-start;
      margin: 5px;
      padding: 5px 10px;
      border-radius: 3px;
      color: var(--ds-text);
      width: 100%;
    }
  }
}

```

File: src/assets/styles/main.scss [hunk 20]
```diff
@@ -23,3 +23,4 @@
 @import "components/workspace-content";
 @import "components/modal";
 @import "components/task-details";
+@import "components/sidebar-taskdetails";
```
Comment (tomercarolla) [inline src/assets/styles/main.scss:26]: sidebar-task-details

File: src/components/task-details/TaskDetails.jsx (hunk 11 above)
Comment (tomercarolla) [reply src/components/task-details/TaskDetails.jsx:92]: @JonathanYesh279

File: src/components/board/Board.jsx [hunk 21]
```diff
@@ -223,6 +224,7 @@ export function Board() {
                     groupId={group.id}
                     title={group.title}
                     tasks={tasks}
+                    board={board}
```
Comment (tomercarolla) [inline src/components/board/Board.jsx:227]: you dont need to pass board here.. is prop drilling. thats why we use redux, get the board in column component from the store.

File: src/components/board/components/Column.jsx [hunk 22]
```diff
@@ -14,7 +14,7 @@ import invariant from 'tiny-invariant';
 import { NewTask } from './NewTask.jsx';
 import { Task } from './Task.jsx';
 
-export const Column = ({ groupId, title, tasks }) => {
+export const Column = ({ board, groupId, title, tasks }) => {
```
Comment (tomercarolla) [inline src/components/board/components/Column.jsx:17]: remove board prop

File: src/components/board/components/Column.jsx [hunk 23]
```diff
@@ -128,7 +128,12 @@ export const Column = ({ groupId, title, tasks }) => {
             <div className="dynamic-content">
               <div className="tasks-list">
                 {tasks.map((task) => (
-                  <Task key={task.id} {...task} />
+                  <Task
+                    key={task.id}
+                    {...task}
+                    groupId={groupId}
+                    board={board}
```
[the remark this answers is not in the document]
Comment (tomercarolla) [reply src/components/board/components/Column.jsx:135]: also get the groupId from the context. remove prop

File: src/components/board/components/Task.jsx [hunk 24]
```diff
@@ -10,10 +10,10 @@ import {
 import { Button, Icon, Modal } from '@ui';
 import { useEffect, useRef, useState } from 'react';
 import invariant from 'tiny-invariant';
-import { TaskDetails } from '../../taskDetails/TaskDetails.jsx';
+import { TaskDetails } from '../../task-details/TaskDetails.jsx';
 import { useBoardContext } from '../board-context.jsx';
 
-export const Task = ({ ...task }) => {
+export const Task = ({ board, groupId, ...task }) => {
```
Comment (tomercarolla) [inline src/components/board/components/Task.jsx:16]: also get the groupId from the context. remove prop

File: src/components/board/components/Task.jsx [hunk 25]
```diff
@@ -91,10 +91,9 @@ export const Task = ({ ...task }) => {
     <Modal
       open={modalOpen}
       onOpenChange={setModalOpen}
-      title="Task Details"
       trigger={taskContent}
     >
-      <TaskDetails task={task} />
+      <TaskDetails task={task} groupId={groupId} board={board} />
```
Comment (tomercarolla) [inline src/components/board/components/Task.jsx:96]: also prop drilling. please fix it

File: src/components/task-details/TaskDetails.jsx [hunk 26]
```diff
@@ -0,0 +1,135 @@
+import { Button, Icon } from '@ui';
+import { useEffect, useState } from 'react';
+import { updateTask } from '../../store/board/board.actions';
+import { NavTaskDetails } from './components/NavTaskDetails';
+
+
+export function TaskDetails({ task, groupId, board }) {
```
Comment (tomercarolla) [inline src/components/task-details/TaskDetails.jsx:7]: remove groupId and board. use context or store

File: src/assets/styles/components/task-details.scss [hunk 27]
```diff
@@ -1,5 +1,185 @@
...
+          display: flex;
+          align-items: 'center';
+          justify-content: 'space-between';
+          gap: '10px';
+        }
+
+        .textarea-description {
+          background-color: var('--surface');
+        }
+        margin-top: 20px;
```
Comment (tomercarolla) [inline src/assets/styles/components/task-details.scss:113]: if the margin-top:20px is to the class description-container, move before the other classes

File: src/components/task-details/TaskDetails.jsx [hunk 28]
```diff
@@ -0,0 +1,191 @@
...
+  const groupTitle = board.groups[groupId]?.title || 'Unknown List';
+  
+  useEffect(() => {
+    setTitle(task.title);
+    setDescription(task.description || '');
+  }, [task]);
+
+  useEffect(() => {
+    if (showTitleInput && inputRef.current) {
+      inputRef.current.focus();
```
Comment (tomercarolla) [inline src/components/task-details/TaskDetails.jsx:26]: add also inputRef.current.select()

File: src/components/task-details/TaskDetails.jsx [hunk 29]
```diff
@@ -0,0 +1,191 @@
...
+              >
+                {title || <span>Click to add a title</span>}
+              </div>
+            )}
+          </div>
+          <div className="group-container">
+            <p>
+              in list:
+              <Button scale="neutral" size="sm" className="btn">
+                <span>{groupTitle}</span>
```
Comment (tomercarolla) [inline src/components/task-details/TaskDetails.jsx:83]: dont need the span, if you dont have icon

File: src/components/task-details/TaskDetails.jsx [hunk 30]
```diff
@@ -0,0 +1,191 @@
...
+                  <Icon name="plus" size="16px" />
+                </Button>
+              </div>
+            </div>
+
+            <div className="labels">
+              <span>Labels</span>
+              <div>
+                <Button scale="neutral" className="btn">
+                  Label
```
Comment (tomercarolla) [inline src/components/task-details/TaskDetails.jsx:107]: what is this button? if here is the labels run from the task db

File: src/components/task-details/components/NavTaskDetails.jsx [hunk 31]
```diff
@@ -0,0 +1,62 @@
...
+        key={name}
+        className="btn-container"
+        style={{ position: 'relative' }}
+      >
+        <Menu trigger={trigger} title={text}>
+          {MenuContent ? <MenuContent /> : <div>No Content for {text}</div>}
+        </Menu>
+      </div>
+    );
+  }
```
Comment (tomercarolla) [inline src/components/task-details/components/NavTaskDetails.jsx:53]: move to new file, it looks like a component, give a better name instead renderButton

File: src/ui/Menus/LabelMenu.jsx [hunk 32]
```diff
@@ -0,0 +1,110 @@
...
+      </StyledDiv>
+      <List>
+        <li>
+          <LabelWrapper>
+            <Checkbox type="checkbox" />
+            <Label style={{ backgroundColor: '#216e4e' }}></Label>
+            <Button scale="ghost" style={{ color: 'var(--ds-text)' }}>
+              <Icon name="edit" size="16px" />
+            </Button>
+          </LabelWrapper>
```
Comment (tomercarolla) [inline src/ui/Menus/LabelMenu.jsx:22]: create a component for this, you repeat yourself too much

File: src/ui/Menus/LabelMenu.jsx [hunk 33]
```diff
@@ -0,0 +1,110 @@
...
+
+const LabelWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  width: 100%;
+  margin-bottom: 5px;
+  gap:5px;
+`
+const Checkbox = styled.input`
+`
```
Comment (tomercarolla) [inline src/ui/Menus/LabelMenu.jsx:67]: remove

**Review by tomercarolla — APPROVED** [review]

**Pull request merged**